src/EnergyPlus/SolarShading.cc

```diff
@@ -7472,7 +7317,7 @@ namespace SolarShading {
                                         } else {
                                             if (state.dataConstruction->Construct(ConstrNumBack).TransDiff <= 0.0) {
                                                 // Do not take into account this window if it is scheduled for surface gains
-                                                SurfSolIncPtr = SurfaceScheduledSolarInc(BackSurfaceNumber, ConstrNumBack);
+                                                int SurfSolIncPtr = SurfaceScheduledSolarInc(BackSurfaceNumber, ConstrNumBack);
```

Comment:
`SurfSolIncPtr` can be set at the top of this loop at line 7246 and then used in both if statements. And it's tempting to add a global bool for `AnyScheduledSurfaceSolar` to avoid the extra steps in `int SurfaceScheduledSolarInc` when there aren't any.

Reply:
Fixed.